be1-go/channel/election/mod.go

```diff
@@ -500,13 +503,39 @@ func checkMethodProperties(method string, length int) error {
 	return nil
 }
 
-func changeVote(qs *question, earlierVote validVote, sender string, created int64, indexes []int) {
-	if earlierVote.voteTime > created {
-		qs.validVotes[sender] = validVote{
-			voteTime: created,
-			indexes:  indexes,
+func updateVote(msgID string, sender string, castVote messagedata.VoteCastVote, questions *map[string]*question) error {
+	// this should update any previously set vote if the message ids are the same
+	for _, vote := range castVote.Votes {
+
+		qs, ok := (*questions)[vote.Question]
+		if !ok {
+			return answer.NewErrorf(-4, "no Question with question ID %s exists", vote.Question)
+		}
+
+		// this is to handle the case when the organizer must handle multiple
+		// votes being cast at the same time
+		qs.validVotesMu.Lock()
+		earlierVote, ok := qs.validVotes[sender]
+
+		// if the sender didn't previously cast a vote or if the vote is no
+		// longer valid update it
+		if err := checkMethodProperties(qs.method, len(vote.Vote)); err != nil {
```

Comment:
This is opiniated but we neved use assignment in if statments. That makes the code harder to read.